tower-oauth2-resource-server/src/validation.rs

```diff
@@ -1,19 +1,52 @@
-use std::fmt::Display;
+use std::{collections::HashSet, fmt::Display};
 
-#[derive(Clone, Debug, Default, Eq, PartialEq)]
+use jsonwebtoken::Algorithm;
+
+/// Returns the default set of allowed algorithms for JWT validation.
+///
+/// This includes all standard asymmetric algorithms that are considered secure.
+/// HMAC algorithms (HS256, HS384, HS512) are excluded by default as they are
+/// symmetric and typically not appropriate for OAuth2/OIDC flows where the
+/// authorization server and resource server are separate entities.
+pub fn default_allowed_algorithms() -> HashSet<Algorithm> {
```

Comment:
In its current state the purpose of `ClaimsValidationSpec` is to state what claims to validate. I think that allowed algorithms is something different and will probably suit better in the `TenantConfiguration` struct.

What do you think?